### PR TITLE
CAMEL-19113 Platform-http-vertx: consume with comma separated values does not work

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://issues.apache.org/jira/browse/CAMEL-19113

It would be nice to apply this fix also for `camel-3.18.x` branch.

# Description

# Target

- [ ] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [ ] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [ ] I checked that each commit in the pull request has a meaningful subject line and body.

- [ ] I formatted the code using `mvn -Pformat,fastinstall install && mvn -Psourcecheck`

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by jamesnetherton** [comment]
I guess we should also do the same for `produces`.

**Comment by JiriOndrusek** [comment]
make sense, I'll add the same functionality for produces and cover it with tests.

**Comment by JiriOndrusek** [comment]
Handling of `produces`  (+test) is added. PR could be merged.

File: components/camel-platform-http-vertx/src/main/java/org/apache/camel/component/platform/http/vertx/VertxPlatformHttpConsumer.java [hunk 1]
```diff
@@ -111,10 +111,16 @@ protected void doStart() throws Exception {
...
-            newRoute.consumes(getEndpoint().getConsumes());
+            //comma separated contentTypes has to be registered one by one
+            for (String c : getEndpoint().getConsumes().split(",")) {
+                newRoute.consumes(c);
+            }
         }
         if (getEndpoint().getProduces() != null) {
-            newRoute.produces(getEndpoint().getProduces());
+            //comma separated contentTypes has to be registered one by one
+            for (String c : getEndpoint().getProduces().split(",")) {
```
Comment (davsclaus) [inline components/camel-platform-http-vertx/src/main/java/org/apache/camel/component/platform/http/vertx/VertxPlatformHttpConsumer.java:121]: You may want to call this `p`

**Review by davsclaus — APPROVED** [review]

File: components/camel-platform-http-vertx/src/main/java/org/apache/camel/component/platform/http/vertx/VertxPlatformHttpConsumer.java (hunk 1 above)
Comment (JiriOndrusek) [reply components/camel-platform-http-vertx/src/main/java/org/apache/camel/component/platform/http/vertx/VertxPlatformHttpConsumer.java]: Copy-paste error. It's fixed now.

**Comment by JiriOndrusek** [comment]
PRs: https://github.com/apache/camel/pull/9473, https://github.com/apache/camel/pull/9474, https://github.com/apache/camel/pull/9475

**Pull request merged**